site-src/geps/gep-1762.md

```diff
@@ -0,0 +1,309 @@
+* MUST mark the Gateway as Ready and provide an address in `Status.Addresses` where the Gateway can be reached on each configured port.
+* MUST label all generated resources (Service, Deployment, etc) with `gateway.networking.k8s.io/metadata.name: my-gateway` (where `my-gateway` is the name of the Gateway resource).
+* MUST name all generated resources `my-gateway-example` (`<NAME>-<GATEWAY CLASS>`).
+  This is not simply `NAME` to reduce the chance of conflicts with existing resources.
+
+### Manual Deployments
+
+When a manual deployment is used, a user must explicitly link their Gateway to existing infrastructure. This is done by specifying `Spec.Address`.
+
+For the common case that the "existing infrastructure" is a `Service`:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: my-gateway
+spec:
+  infrastructure:
+    attachTo:
+      kind: Service
+      name: existing-service
+  addresses:
+  - type: Service
+    value: some-existing-service
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+```
+
+With this configuration, an implementation:
+* MUST mark the Gateway as Ready and provide an address in `Status.Addresses` where the Gateway can be reached.
+  This MUST be derived from the referenced `Spec.Address`.
+* MUST not deploy any resources into the cluster; it is expected that a user will do these actions.
+
+### Merging Gateways
+
+The spec currently specifies that gateways *can* be merged.
+If an implementation supports this, it MUST be implemented by referencing the `Gateway` to merge into.
+
+For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: merged-gateway
+spec:
+  infrastructure:
+    attachTo:
+      kind: Gateway
+      name: parent-gateway
+  gatewayClassName: example
+  listeners:
+  - name: some-other-route
+    port: 81
+    protocol: HTTP
+```
+
+With this configuration, an implementation:
+* MUST treat the `parent-gateway` as having the concatenated list of all listeners from merged Gateways.
+  This means, for example, that a Service with port 80 and 81 may be created in this example.
+
+This introduces a new concept in Gateway merging of a "primary" Gateway.
+This makes it explicit which is the primary gateway and which are attached to it.
+
+Secondary Gateways cannot configure `addresses` or anything in `infrastructure` (aside from `attachTo`), and must have
+the same `gatewayClassName` as the primary Gateway they are attached to.
+
+Gateways can only `attachTo` a primary Gateway. That is, we cannot have a chain of references.
+
+### Customizations
+
+With any in-cluster deployment, customization requirements will arise. 
+
+Some common requirements would be:
```

Comment:
To clarify, if we can attach a Gateway to an existing Service, do we need to allow anything else to be configurable on the Gateway, or should we just rely on users provisioning custom Services whenever customization is needed?

Reply:
Good point - a Service of type LoadBalancer with selector and associated deployment ( created by user ) and attached Gateway is a very reasonable model. Gateway can customize the TLS server setting and define routing, but can reuse the Service address and deployment.

Reply:
Yes, we do. There is a big difference between "Manual deployment" and "Customized deployment".

Lets say I want to set `sessionAffinity` on service. First, I need to go create my Service. I guess I know port 443 will be used in the gateway, so I create the Service and link it up to the Gateway.

A week later, I add port 80 to the Gateway as well. Now its broken; I need to keep the Service in sync.

With _customization_, a user would simply configure it wants sessionAffinity, but the implementation would be responsible for keeping everything else in sync.